src/d/parser/expression.d

```diff
@@ -1047,3 +1050,36 @@ StringLiteral parseStringLiteral(ref TokenRange trange) {
 	
 	return new StringLiteral(location, name.toString(trange.context));
 }
+
+/**
+ * Parse floating point literals
+ */
+FloatLiteral parseFloatLiteral(ref TokenRange trange) {
```

Comment:
Honestly, as a first iteration, I'd extract this guy, ass a test for it, get the parser to generate an AST node, done. This part looks quite solid already.

I'm not sure the `to!float` here is a such a great idea. I admit part of me like it, but that also means moving the spec in the standard lib, which seems wrong. What does DMD do?

Reply:
dmd (ldc and gdc are allowed to fiddle with the implementation) just uses whatever the system `strtold` does. Oh and it does everything as real (yay!)

Reply:
@deadalnix Where do I put the test?

Reply:
In a unit test block?

Reply:
But where would you prefer? No unit tests directly in this file, so if parseStringLiteral is tested via the whole parser where does it go. It's probably obvious but I'm away from the Dev box rn

Reply:
You can run them using `make check-libd-parser`

Careful though, DMD doesn't track dependencies properly, so in case you think you might be runnign stale code: `rm -rf /tmp/.rdmd-$[USERID}`